baseplate/lib/timer.py

```diff
@@ -0,0 +1,56 @@
+from enum import Enum
+from enum import unique
+from typing import Callable
+from typing import Optional
+
+from schedule import CancelJob
+
+from baseplate.lib.threaded_scheduler import ThreadedScheduler
+
+
+@unique
+class TimerState(Enum):
+    Running = "Running"
+    Stopped = "Stopped"
+
+
+class Timer:
+    def __init__(
+        self,
+        function: Callable[[], None],
+        interval: float,
+        scheduler_interval: Optional[float] = None,
+        is_repeating: bool = False,
+    ) -> None:
+        self.function = function
+        self.interval = interval
+        self.scheduler_interval = scheduler_interval if scheduler_interval is not None else interval
+        self.is_repeating = is_repeating
+        self.state = TimerState.Stopped
+        self.scheduler = ThreadedScheduler()
+
+    def start(self) -> None:
+        if self.state == TimerState.Running:
+            return
+        self.state = TimerState.Running
+        self.scheduler.every(self.interval).seconds.do(self._invoke)
+        self.scheduler.run_continuously(self.scheduler_interval)
```

Comment:
I don't think there's a method called `run_continuously` did you mean to use `run`?

Reply:
Yup, mistake during refactoring